### DISCUSSION TITLE
Purge All Releases

### DISCUSSION THREAD

**Discussion by sudo-nautilus** [community/community#23376] [New to GitHub]
<p>Ok so, is there any way to purge <strong>all</strong> the releases/tags at once?<br>
I can do that buy deleting them one by one. But it will take so long to do the same for all the 140 releases. That’s why I’m asking. Thank you in advance</p>

**Comment by jsoref** [comment]
<p>I use<br>
<code>empty-remote</code>:</p>
<pre><code class="lang-auto">#!/bin/sh
if [ -z "$1" ]; then
  echo "$0 url-for-remote-git-repository-with-branches-or-tags-to-remove" &gt;&amp;2
  exit 1
fi
if command -v parallel &gt;/dev/null; then
  dispatch() {
    parallel -I% --max-args 25 git push "$1" %
  }
else
  dispatch() {
    xargs -n10 git push "$1"
  }
fi
cd ~/code/empty
git ls-remote "$1" |perl -ne 'next if /HEAD|\{}/;s/^\S+\s+/:/;print'|dispatch "$1"
</code></pre>
<p>note <code>~/code/empty</code> needs to be an empty git repository (<code>git init ~/code/empty</code>).</p>

**Comment by sudo-nautilus** [comment]
<p>I’m pretty nervous about that. But anyway thank you :heart:</p>

**Comment by jsoref** [comment]
<p>:warning: This will kill <strong>all</strong> branches that aren’t the default branch.<br>
:information_source: Add an <code>echo</code> before <code>git push</code> to see what it does.<br>
You can change the <code>next if /.../</code> code if you want to not delete certain other branches.</p>
<p>I use this code frequently (which is why it’s a real script). Basically I fork a repository and then immediately run this script against the fork. But since I’m usually using it against a fork, if I want a branch back, I can pull it from the original repository.</p>

**Comment by sudo-nautilus** [comment]
<p>Would you mind checking <a href="https://github.com/sudo-nautilus/FFmpeg-Builds-Win32" rel="noopener nofollow ugc">this repo</a> ?<br>
And could you please instruct me, how to do that?<br>
Thank you in advance</p>

**Comment by tajmone** [comment]
<p>Destroying all tags (locally) and then updating your origin accordingly should do the trick without deleting any branches — GH releases are tag-based, so no tags no releases. Just ensure that your local close is set to track tags too.</p>

**Comment by sudo-nautilus** [comment]
<aside class="quote no-group" data-username="tajmone" data-post="6" data-topic="190385">
<div class="title">
<div class="quote-controls"></div>
 tajmone:</div>
<blockquote>
<p>Destroying all tags (locally) and then updating your origin accordingly should do the trick</p>
</blockquote>
</aside>
<p>I’ll try it :slight_smile:</p>

**Comment by sudo-nautilus** [comment, marked as answer]
<p>That worked actually,<br>
I ran this command:</p>
<pre><code class="lang-bash">git tag | xargs git push origin --delete
</code></pre>

**Comment by sudo-nautilus** [comment]
<p>A great thanks to <a class="mention" href="/jsoref">@jsoref</a> and <a class="mention" href="/tajmone">@tajmone</a> :wink:<br>
I’ll mark my reply as an answer to the topic.</p>
